File: ios/NumberSumModule.m
//
//  NumberSumModule.m
//  BridgeDemo
//
//  Created by Neosoft on 10/01/24.
//

//#import <Foundation/Foundation.h>
#import "NumberSumModule.h"

@implementation NumberSumModule

// To export a module named NumberSumModule
RCT_EXPORT_MODULE();

RCT_EXPORT_METHOD(calculateSum:(double)num1 withNumber2:(double)num2 resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject) {
//  double sum = num1 + num2 + 10;
//  resolve([NSNumber numberWithDouble:sum]);
  @try {
      // Perform your calculation here
      double sum = num1 + num2;
      
      // Resolve the promise with the result
      resolve([NSNumber numberWithDouble:sum]);
    } @catch (NSException *exception) {
      // If an exception occurs, reject the promise with an error code and message
      NSString *errorMessage = [NSString stringWithFormat:@"Error performing calculation: %@", exception.reason];
      NSError *error = [NSError errorWithDomain:@"CALCULATION_ERROR" code:-1 userInfo:@{NSLocalizedDescriptionKey: errorMessage}];
      reject(@"CALCULATION_ERROR", errorMessage, error);
    }
}

@end

File: android/app/src/main/java/com/bridgedemo/NumberSumModule.kt
package com.bridgedemo;
import android.util.Log
import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod

class NumberSumModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {

    override fun getName(): String {
        return "NumberSumModule"
    }

    @ReactMethod
    fun calculateSum(num1: Double, num2: Double, promise: Promise) {
       Log.d("adding","aaa")
        try {
            val sum = num1 + num2 + 10
            promise.resolve(sum)
        } catch (e: Exception) {
            promise.reject("CALCULATION_ERROR", "Error performing calculation", e)
        }
    }
}